.phpcs.xml.dist

```diff
@@ -0,0 +1,215 @@
+<?xml version="1.0"?>
+<ruleset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	name="Requests Test Server"
+	xsi:noNamespaceSchemaLocation="./vendor/squizlabs/php_codesniffer/phpcs.xsd">
+
+	<description>Requests Test Server rules for PHP_CodeSniffer</description>
+
+	<!--
+	#############################################################################
+	COMMAND LINE ARGUMENTS
+	https://github.com/squizlabs/PHP_CodeSniffer/wiki/Annotated-ruleset.xml
+	#############################################################################
+	-->
+
+	<!-- Scan all files. -->
+	<file>.</file>
+
+	<!-- Third party files and build files don't need to comply with these coding standards. -->
+	<exclude-pattern>*/vendor/*</exclude-pattern>
+
+	<!-- Only check PHP files. -->
+	<arg name="extensions" value="php"/>
+
+	<!-- Show progress, show the error codes for each message (source). -->
+	<arg value="ps"/>
+
+	<!-- Strip the filepaths down to the relevant bit. -->
+	<arg name="basepath" value="./"/>
+
+	<!-- Check up to 8 files simultaneously. -->
+	<arg name="parallel" value="8"/>
+
+
+	<!--
+	#############################################################################
+	CHECK FOR PHP CROSS-VERSION COMPATIBILITY
+	#############################################################################
+	-->
+
+	<config name="testVersion" value="5.6-"/>
+	<rule ref="PHPCompatibility"/>
+
+
+	<!--
+	#############################################################################
+	CODING STYLE RULES
+	#############################################################################
+	-->
+
+	<!-- Use the WordPress Coding Standards as a basis, but with tweaks. -->
+	<rule ref="WordPress-Extra">
```

Comment:
We discussed keeping the code style consistent across repositories last time, but now that I'm seeing the relative size of the code style rules VS the actual code, I'm wondering whether we shouldn't ignore consistency with the main library here, because:
- this code is minimal and has very little to do with the actual library
- we don't expect the same contributors to jump in here as well
- just sticking with something like PSR-12 would get rid of a lot of the custom style rules AND remove the WPCS dependency

Reply:
Happy to change the PR to PSR12 + some extra rules instead. Will need to wait for our next session though as I won't have time to do this now.

Reply:
Updated now anyway:
* PR has been rebased after #12 was merged.
* I've created a new PHPCS ruleset based on PSR-12 with a few tweaks (6d5b962693712a14f265b912eebeebb5bee9d1d2)
    The most important tweak is probably enforcing tab indentation instead of space indentation. I did so to keep the code churn lower as otherwise every single line in the code base would have needed to be touched.
* I've redone the fixes to comply with the new ruleset (+ some extra).
* I've updated the PR description to match the new commits.

Note: the change which moved the `parse_params_rfc()` function to a separate file is no longer needed now, so has been removed.